components/org.wso2.carbon.identity.branding.preference.management.core/src/test/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImplTest.java

```diff
@@ -369,6 +373,289 @@ public void testDeleteNotExistingBrandingPreference(String tenantDomain, int ten
+        Assert.assertEquals(retrievedCT.getScreen(), inputCT.getScreen());
+
+        // Deleting added custom text preference.
+        brandingPreferenceManagerImpl.deleteCustomText(inputCT.getType(), inputCT.getName(), inputCT.getScreen(),
+                inputCT.getLocale());
+    }
+
+    @DataProvider(name = "invalidCustomTextPreferenceDataProvider")
+    public Object[][] invalidCustomTextPreferenceDataProvider() throws IOException {
+
+        CustomText customText1 = new CustomText();
+        customText1.setType(ORGANIZATION_TYPE);
+        customText1.setName(SUPER_TENANT_DOMAIN_NAME);
+        customText1.setScreen(LOGIN_SCREEN);
+        customText1.setLocale(DEFAULT_LOCALE);
+        customText1.setPreference(1234);
+
+        CustomText customText2 = new CustomText();
+        customText2.setType(ORGANIZATION_TYPE);
+        customText2.setName(SUPER_TENANT_DOMAIN_NAME);
+        customText2.setScreen(LOGIN_SCREEN);
+        customText2.setLocale(DEFAULT_LOCALE);
+        customText2.setPreference("Branding Preference");
+
+        CustomText customText3 = new CustomText();
+        customText3.setType(ORGANIZATION_TYPE);
+        customText3.setName(SUPER_TENANT_DOMAIN_NAME);
+        customText3.setScreen(LOGIN_SCREEN);
+        customText3.setLocale(DEFAULT_LOCALE);
+        customText3.setPreference(new JSONObject());
+
+        return new Object[][]{
+                {customText1},
+                {customText2},
+                {customText3},
+        };
+    }
+
+    @Test(dataProvider = "invalidCustomTextPreferenceDataProvider")
+    public void testAddInvaliCustomTextPreference(Object customText) {
+
+        CustomText inputCT = (CustomText) customText;
+
+        // Adding new custom text preference.
+        assertThrows(BrandingPreferenceMgtClientException.class, () -> brandingPreferenceManagerImpl
+                .addCustomText(inputCT));
+    }
+
+    @Test(dataProvider = "customTextPreferenceDataProvider")
+    public void testGetCustomTextPreference(Object customText, String tenantDomain, int tenantId)
+            throws Exception {
+
+        setCarbonContextForTenant(tenantDomain, tenantId);
+        CustomText inputCT = (CustomText) customText;
+
+        // Adding new custom text preference.
+        CustomText addedCT = brandingPreferenceManagerImpl.addCustomText(inputCT);
+
+        // Retrieving added custom text preference.
+        CustomText retrievedCT = brandingPreferenceManagerImpl.getCustomText
+                (inputCT.getType(), inputCT.getName(), inputCT.getScreen(), inputCT.getLocale());
+        Assert.assertEquals(retrievedCT.getPreference(), inputCT.getPreference());
+        Assert.assertEquals(retrievedCT.getName(), inputCT.getName());
+        Assert.assertEquals(retrievedCT.getType(), inputCT.getType());
+        Assert.assertEquals(retrievedCT.getScreen(), inputCT.getScreen());
+        Assert.assertEquals(retrievedCT.getLocale(), inputCT.getLocale());
```

Comment:
seems this is redundant in get method's unit tests. keep one

Reply:
Adddressed with https://github.com/wso2-extensions/identity-branding-preference-management/commit/fdd3486cfdc51558c723795cf2fb75b6faafcdc5

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/14/commits/780ee9124673551a420d76452dd3d3ed0ee1ee5d